### ISSUE TITLE
Implement new Code Generator

### ISSUE THREAD

**Issue by ceeac** [BoomerangDecompiler/boomerang#112]
The current Code Generator does not produce very readable code, especially not for very complex functions. We should implement a new code generator e.g. based on the "[No More Gotos](https://net.cs.uni-bonn.de/fileadmin/ag/martini/Staff/yakdan/dream_ndss2015.pdf)" paper. For this, the current Control Flow analysis needs to be refactored to analyze Statements instead of BasicBlocks.

**Comment by rfalke** [comment]
This sounds like a big task. Since I think the rewrite for capstone (https://github.com/BoomerangDecompiler/boomerang/pull/69) is also a bit task I would suggest to finish one first.

**Comment by ceeac** [comment]
Yes; the roadmap I outlined in [this comment](https://github.com/BoomerangDecompiler/boomerang/issues/68#issuecomment-417843133) still stands. So while #69 will be merged more or less directly after the release of 0.4.0, the new Code Generator is much farther into the future - probably after 0.5.0 (unless somebody implements it beforehand).
